src/main/java/life/qbic/model/download/QbicDataDownloader.java

```diff
@@ -187,134 +193,185 @@ private void downloadDatasets(Sample analyte, List<DataSet> dataSets,
               + datasetSample + " (" + permID + ")");
-      downloadFiles(downloadRequest);
+      downloadResponse.merge(downloadFiles(downloadRequest));
     }
+    return downloadResponse;
   }
 
-  private void downloadFile(DataSetFile dataSetFile, Path prefix) throws DownloadException {
+  private FileDownloadResponse downloadFile(DataSetFile dataSetFile, Path prefix) throws DownloadException {
     for (IDataStoreServerApi dataStoreServer : dataStoreServers) {
       try {
-        downloadFileFromDataStoreServer(dataSetFile, prefix, dataStoreServer);
-        break;
+        return downloadFileFromDataStoreServer(dataSetFile, prefix, dataStoreServer);
       } catch (FileNotFoundException fileNotFoundException) {
         // log and try the next data store server
         LOG.debug("Download attempt failed on " + dataStoreServer, fileNotFoundException);
-        continue;
       }
     }
+    return FileDownloadResponse.failure();
   }
 
+  // note: DataSetFileDownloadReader closes the input stream after it finished reading it.
+  private static class AutoClosableDataSetFileDownloadReader extends DataSetFileDownloadReader implements AutoCloseable {
+    public AutoClosableDataSetFileDownloadReader(InputStream in) {
+      super(in);
+    }
+  }
+
+  public static class FileDownloadResponse {
+    private final boolean success;
+
+    private FileDownloadResponse(boolean success) {
+      this.success = success;
+    }
+
+    public static FileDownloadResponse success() {
+      return new FileDownloadResponse(true);
+    }
+
+    private static FileDownloadResponse failure() {
+      return new FileDownloadResponse(false);
+    }
+
+    public boolean isSuccess() {
+      return success;
+    }
+
+    public boolean isFailure() {
+      return !success;
+    }
+  }
+
+
   /**
-   * @throws FileNotFoundException if the file was not found on the data store server.
+   * @throws FileNotFoundException if the file was not on the data store server.
    */
-  private void downloadFileFromDataStoreServer(DataSetFile dataSetFile, Path prefix,
+  private FileDownloadResponse downloadFileFromDataStoreServer(DataSetFile dataSetFile, Path prefix,
       IDataStoreServerApi dataStoreServerApi) throws FileNotFoundException {
 
+    // skip if file is empty
+    if (dataSetFile.getFileLength() < 1) {
+      LOG.info("Skipped empty file " + dataSetFile.getPath());
+      return FileDownloadResponse.success();
+    }
+
+    Path localFilePath = OutputPathFinder.determineOutputDirectory(outputPath, prefix, dataSetFile,
+        conservePaths);
+    File localFile = localFilePath.toFile();
+
+    // skip if same content exists already
+    if (localFile.exists()) {
+      System.out.print("Found existing file. Checking content...");
+
+      if (localFileWithSameContent(dataSetFile, localFile)) {
+        System.out.println("\rFound existing file with identical content. Skipping "
```

Comment:
Logging?

Reply:
`localFileWithSameContent` does the logging of the relevant information. The sout calls are only of interest at the moment to perceive progress.